src/components/App.tsx

```diff
@@ -918,7 +913,7 @@ export class App extends React.Component<any, AppState> {
           };
         });
       });
-      this.socket.on("new-user", async (socketID: string) => {
+      this.socket.on("new-user", async (_socketID: string) => {
```

Comment:
we should add shadowing in our ESLint rules... it's no fun.. but has to happen!

Reply:
https://eslint.org/docs/rules/no-shadow

Reply:
Why? What does it solve right now?

Reply:
In my case, tsc warns unused vars, and it was hard to grasp other errors. Is it only for me?

Reply:
So, if we would detect it with eslint, it's https://eslint.org/docs/rules/no-unused-vars.
`no-shadow` would warns the current codebase way too many.

Reply:
> Why? What does it solve right now?

Right now.. nothing.. in the long run, something :)

Reply:
> Right now.. nothing.. in the long run, something :)

I vote against this. Until this creates bugs, let's not do this. It'd force us to start prefixing `_` to variable names or having to come up with random names just to satisfy the linter.

> @dai-shi  In my case, tsc warns unused vars

And that was actually correct → the parameter is unused and should be removed. I have no idea why vscode nor TS doesn't warn us when prefix it as `_socketID`. In fact, without the the prefix, only vscode warns and eslint ignores it, too :thinking:.

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#flag-unused-declarations-with---nounusedparameters-and---nounusedlocals

> Parameters declaration with names starting with _ are exempt from the unused parameter checking.

Reply:
Good catch. Don't know why that's the case, but another reason for not using `no-shadow` rule.

Reply:
It's not only to satisfy the linter.. no-shadow rule is a well known rule in programming! Doesn't bite all the time.. but it can :) Better to be avoided.. not sure why you guys are afraid of setting some rules.. this codebase is growing fast and we have to make sure it stays sane and if robots can help us, that's a plus..

Reply:
@lipis To be clear, this change in question is for "no-unused" rule (in this case, TS not eslint) not "no-shadow" rule.

That said, I'm not against to no-shadow rule in general, and most of my projects enable that rule. But because the current codebase in this project is violating "no-shadow" in various places, it's not just about enabling the rule. It's not nice to just rename variables (it would become like `x`, `xx`, `xxx`...), but this requires refactoring (more code split into smaller functions). Is it in line with the discussion?

Reply:
Eventually we'll have to split the code.. some files are getting too large.. yes naming x, xx, xxx is not a good idea, but something like `updated_x`, `rotated_x`, etc.. makes more sense, and we don't have to be super short on the variable names.. there is enough space!

That's for another discussion/PR though.